webflow-hacks/misc-hacks/15/15.ts

```diff
@@ -0,0 +1,29 @@
+// when the DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  const MENU_BUTTON_SELECTOR = '[fs-hacks-element="menu-button"]';
+  // get the anchor link buttons
+  const menuBtns = document.querySelectorAll(MENU_BUTTON_SELECTOR);
+  if (!menuBtns) return;
+  // for each menu button
+  menuBtns.forEach(function (menuBtn) {
+    // when each button is clicked
+    menuBtn.addEventListener('click', function () {
+      // set a short timeout before taking action
+      // so as to allow hash to be set
+      setTimeout(() => {
+        // call removeHash function after set timeout
+        removeHash();
+      }, 5); // 5 millisecond timeout in this case
```

Comment:
No need to wait 5 miliseconds.
By creating the timeout without any delay you're already moving the callback at the end of the event loop.
So basically this would work:
```typescript
setTimeout(removeHash);
```

Reply:
awesome